ViAn/Test/filehandlertest.cpp

```diff
@@ -4,76 +4,257 @@
  * @brief filehandlertest::filehandlertest
  * @param parent
  */
-filehandlertest::filehandlertest(QObject *parent) : QObject(parent){
+FileHandlerTest::FileHandlerTest(QObject *parent) : QObject(parent){
+    file_handler = new FileHandler();
 
 }
+
+/**
+ * @brief FileHandlerTest::project_test_init
+ * creates project for project tests.
+ */
+void FileHandlerTest::project_test_init()
+{
+    this->proj = file_handler->create_project("TEST_PROJ");
+}
+
+/**
+ * @brief FileHandlerTest::project_create_test
+ * Test creating and deleting project.
+ */
+void FileHandlerTest::project_create_delete_test()
+{
+    Project* proj1 = file_handler->create_project("TEST_PROJ1");
+    Project* proj2 = file_handler->create_project("TEST_PROJ2");
+    Project* proj3 = file_handler->create_project("TEST_PROJ3");
+    QVERIFY(file_handler->delete_project(proj1->id));
+    QVERIFY(file_handler->delete_project(proj2->id));
+    QVERIFY(file_handler->delete_project(proj3->id));
+}
+
+/**
+ * @brief FileHandler::project_save_load_test
+ * Tests saving and loading projects.
+ */
+void FileHandlerTest::project_save_load_test(){
+    Project* proj1 = file_handler->create_project("TEST_PROJ1");
+    Project* proj2 = file_handler->create_project("TEST_PROJ2");
+    Project* proj3 = file_handler->create_project("TEST_PROJ3");
+
+    file_handler->save_project(proj1->id);
+    file_handler->save_project(proj2->id);
+    file_handler->save_project(proj3->id);
+
+    Project* l_proj1 = file_handler->load_project(file_handler->get_dir(proj1->dir).absoluteFilePath("TEST_PROJ1.json").toStdString());
+    Project* l_proj2 = file_handler->load_project(file_handler->get_dir(proj2->dir).absoluteFilePath("TEST_PROJ2.json").toStdString());
+    Project* l_proj3 = file_handler->load_project(file_handler->get_dir(proj3->dir).absoluteFilePath("TEST_PROJ3.json").toStdString());
+
+    QVERIFY(file_handler->proj_equals(*proj1,*l_proj1));
+    QVERIFY(file_handler->proj_equals(*proj2,*l_proj2));
+    QVERIFY(file_handler->proj_equals(*proj3,*l_proj3));
+
+    QVERIFY(file_handler->delete_project(proj1->id));
+    QVERIFY(file_handler->delete_project(proj2->id));
+    QVERIFY(file_handler->delete_project(proj3->id));
```

Comment:
Didn't you already test deleting projects in the previous test? You shouldn't need to do it again. Also didn't you have two save_projects, then maybe you should test both of them.